src/App.js

```diff
@@ -1,26 +1,78 @@
-import React, { Component } from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { FindMovie } from './components/FindMovie';
 import data from './api/movies.json';
+import { request } from './api/api';
 
-export class App extends Component {
-  state = {
-    movies: data,
+export const App = () => {
+  const [movies, setMovies] = useState(data);
+  const [query, setQuery] = useState('');
+  const [film, findFilm] = useState(null);
+  const [error, findError] = useState(false);
+
+  const handleSearch = () => {
+    request(query)
+      .then((result) => {
+        if (result.Response === 'False') {
+          findFilm(null);
+
+          return findError(true);
+        }
+
+        return findFilm(result);
+      });
   };
 
-  render() {
-    const { movies } = this.state;
-
-    return (
-      <div className="page">
-        <div className="page-content">
-          <MoviesList movies={movies} />
-        </div>
-        <div className="sidebar">
-          <FindMovie />
-        </div>
+  const handleChange = (event) => {
+    setQuery(event.target.value);
+    findError(false);
+  };
+
+  const addFilm = (movie) => {
+    if (!movie) {
+      return findError(true);
+    }
+
+    if (movies.some(cinema => cinema.imdbId === movie.imdbID) === true) {
+      setQuery('');
+
+      return findFilm(null);
+    }
+
+    const { Title, Plot, Poster, imdbID } = movie;
+    const result = {
+      title: Title,
+      description: Plot,
+      imgUrl: Poster,
+      imdbUrl: `https://www.imdb.com/title/${imdbID}`,
+      imdbId: imdbID,
+    };
+
+    setMovies(array => [...array, result]);
+    setQuery('');
+
+    return findFilm(null);
+  };
+
+  return (
+    <div className="page">
+      <div className="page-content">
+        <MoviesList
+          movies={movies}
+        />
+      </div>
+      <div className="sidebar">
+        <FindMovie
+          film={film}
+          setQuery={setQuery}
+          query={query}
+          handleSearch={handleSearch}
+          error={error}
+          handleChange={handleChange}
+          addFilm={addFilm}
```

Comment:
```suggestion
          addFilm={addFilm}
```
You should only pass the function for adding the movie. All search should be handled inside `FindMovie` component. Other variables and functions are not used in the `App` component